File: src/main/java/com/example/eventplanner/model/Argument.java
package com.example.eventplanner.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

@Entity
@Getter
@Setter
public class Argument {

    public enum ArgumentType {
        PRO, CONTRA
    }

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Enumerated(EnumType.STRING)
    private ArgumentType type;

    private String content;

    @ManyToOne
    @JsonIgnore
    private Topic topic;

    @ManyToOne
    @JoinColumn(name = "user_id")
    private User user;
}

File: frontend/src/app/entities/Argument.ts
// src/app/entities/argument.ts

import {Topic} from "./Topic";
import {User} from "./user";

export enum ArgumentType {
  PRO = 'PRO',
  CONTRA = 'CONTRA'
}

export interface Argument {
  id?: number;
  topic: Topic;  // ID-ul topicului asociat acestui argument
  content: string;
  type: ArgumentType;
}
